tests/components/motioneye/test_config_flow.py

```diff
@@ -287,3 +339,34 @@ async def test_duplicate(hass: HomeAssistant) -> None:
     assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
     assert result["reason"] == "already_configured"
     assert mock_client.async_client_close.called
+
+
+async def test_hassio_already_configured(hass: HomeAssistant) -> None:
```

Comment:
Should we also test that the discovery flow is aborted if there's already a user flow in progress and that an existing discovery flow is clean up if we finish and create an entry for a a user flow?

Reply:
We could, however, that is handled by (and extensively tested by) `_async_handle_discovery_without_unique_id`

Reply:
Yes, but if we remove the use of that helper or replace it with something else, we want this to be tested for this integration and fail if it doesn't work.

Reply:
Added new tests to cover that, discovered a bug in our flow handling using it...
Fixed in #51105. This PR needs a rebase after that has been merged.